src/basic/socket-util.c

```diff
@@ -957,11 +957,11 @@ int getpeercred(int fd, struct ucred *ucred) {
 
         /* Check if the data is actually useful and not suppressed due
          * to namespacing issues */
-        if (u.pid <= 0)
+        if (!pid_is_valid(u.pid))
                 return -ENODATA;
-        if (u.uid == UID_INVALID)
+        if (!uid_is_valid(u.uid))
```

Comment:
Hmm, I'm a bit wary of this. `uid_is_valid` is used in two ways. In the first case to determine if we should allow a certain uid, in which case we should be stringent. In the second case to determine if we got an answer from the kernel. And I think we should not filter values out, unless the kernel interface declares them as placeholders. socket(7) says that "The returned credentials are those that  were in  effect  at  the time of the call to connect(2) or socketpair(2).". This means that they should be valid, always. For example, let's say I create a user with UID=0xFFFF. I don't think our library calls in sd-login should refuse to work on this user.

Reply:
My thinking is that we should refuse early on what we cannot process later. getpeercred() and friends is most prominently used for authenticating data, and hence I'd be very conservative in perrmitting data that we most likely will choke on later, it's just too sensitive, for very little gain...